src/assets/wise5/authoringTool/peer-grouping/edit-peer-grouping-dialog/edit-peer-grouping-dialog.component.ts

```diff
@@ -25,14 +31,63 @@ export class EditPeerGroupingDialogComponent extends AuthorPeerGroupingDialogCom
   ngOnInit(): void {
     this.peerGrouping = this.utilService.makeCopyOfJSONObject(this.peerGrouping);
     this.stepsUsedIn = this.peerGroupingAuthoringService.getStepsUsedIn(this.peerGrouping.tag);
+    this.logicType = this.getLogicType(this.peerGrouping.logic);
+    if (this.logicType === DIFFERENT_IDEAS_VALUE) {
+      this.referenceComponent = this.getDifferentIdeasReferenceComponent(this.peerGrouping.logic);
+    }
+  }
+
+  private getLogicType(logic: string): string {
+    if (new RegExp(DIFFERENT_IDEAS_REGEX).exec(logic) != null) {
+      return DIFFERENT_IDEAS_VALUE;
+    } else {
+      return logic;
+    }
```

Comment:
Use ternary to reduce complexity?
```suggestion
    return new RegExp(DIFFERENT_IDEAS_REGEX).exec(logic) != null ? DIFFERENT_IDEAS_VALUE : logic;
```